process_results.py

```diff
@@ -2,71 +2,63 @@
 
 import argparse
 import json
-import os
+from pathlib import Path
 import re
 
 
-def load_test_names(filename):
-    test_names = []
-    with open(filename) as f:
-        for line in f:
-            m = re.search("RUN_TEST\((.*)\)", line)
-            if m:
-                test_names.append(m.group(1))
-    return test_names
+rgx_testnames = re.compile(r"RUN_TEST\(([^)]+)\);")
+last_test_name = {}
 
 
-def process_results(filename):
+def get_last_test_name(filepath):
+    if filepath not in last_test_name:
+        last_test_name[filepath] = rgx_testnames.findall(
+            Path(filepath).read_text())[-1]
+    return last_test_name[filepath]
+
+
+def truncate(text, maxlength=500):
+    if len(text) > maxlength:
+        text = f"{text[:500]}\nOutput was truncated. Please limit to {maxlength} chars."
```

Comment:
If we specify `maxlength` to something other than 500, the text will still be truncated to 500. Shouldn't it be truncated to `maxlength`?

Reply:
Good catch; that's what I get for copy-pasting :sweat_smile: